include/vara/Feature/FeatureSourceRange.h

```diff
@@ -72,70 +72,72 @@ class FeatureSourceRange {
   };
 
   class FeatureMemberOffset {
-    FeatureMemberOffset(std::string MemberOffset)
-        : MemberOffset(std::move(MemberOffset)) {}
-
   public:
-    static llvm::Optional<FeatureMemberOffset>
-    createFeatureMemberOffset(std::string MemberOffset) {
-      // Member offsets need to have excaptly one :: separator
-      if (isInMemberOffsetFormat(MemberOffset)) {
-        return FeatureMemberOffset(std::move(MemberOffset));
+    static std::optional<FeatureMemberOffset>
+    createFeatureMemberOffset(llvm::StringRef MemberOffset) {
+      auto RSplit = MemberOffset.rsplit("::");
+      if (RSplit.second.empty()) {
+        // wrong format
+        return std::nullopt;
       }
-
-      return {};
-    }
-
-    static bool isInMemberOffsetFormat(llvm::StringRef PossibleMemberOffset) {
-      return PossibleMemberOffset.contains("::") &&
-             PossibleMemberOffset.count(':') == 2;
+      llvm::SmallVector<llvm::StringRef, 1> Class;
+      RSplit.first.split(Class, "::");
+      return FeatureMemberOffset(Class, RSplit.second);
     }
 
-    void setMemberOffset(std::string MemberOffset) {
-      if (isInMemberOffsetFormat(MemberOffset)) {
-        this->MemberOffset = std::move(MemberOffset);
+    [[nodiscard]] std::string
+    className(std::optional<size_t> Nested = std::nullopt) const {
+      if (Nested.has_value()) {
+        assert(Nested.value() < Class.size());
+        return Class[Class.size() - Nested.value() - 1];
       }
-    }
-    [[nodiscard]] std::string fullMemberOffset() const {
-      return this->MemberOffset;
+      std::stringstream StrS;
+      StrS << Class[0];
+      for (size_t Idx = 1; Idx < Class.size(); ++Idx) {
+        StrS << "::" << Class[Idx];
+      }
+      return StrS.str();
     }
 
-    [[nodiscard]] llvm::StringRef className() const {
-      llvm::StringRef TmpRef{MemberOffset};
-      return TmpRef.substr(0, TmpRef.find_first_of(':'));
-    }
+    [[nodiscard]] std::string memberName() const { return Member; }
 
-    [[nodiscard]] llvm::StringRef memberName() const {
-      llvm::StringRef TmpRef{MemberOffset};
-      return TmpRef.substr(TmpRef.find_last_of(':') + 1);
+    [[nodiscard]] std::string toString() const {
+      return className() + "::" + memberName();
     }
 
-    [[nodiscard]] std::string toString() const { return fullMemberOffset(); }
-
     inline bool operator==(const FeatureMemberOffset &Other) const {
-      return fullMemberOffset() == Other.fullMemberOffset();
+      return Member == Other.Member && Class == Other.Class;
     }
 
     inline bool operator!=(const FeatureMemberOffset &Other) const {
-      return fullMemberOffset() != Other.fullMemberOffset();
+      return !(*this == Other);
     }
 
   private:
-    std::string MemberOffset;
+    FeatureMemberOffset(const llvm::SmallVector<llvm::StringRef, 1> &Class,
+                        llvm::StringRef Member)
+        : Member(Member.str()) {
```

Comment:
```suggestion
        : Class{Class.begin(), Class.end()} ,Member(Member.str()) {
```
This correctly transfers the items of the SmallVector